source/common/http/conn_manager_impl.cc

```diff
@@ -821,6 +819,24 @@ void ConnectionManagerImpl::ActiveStream::refreshCachedRoute() {
   cached_route_ = std::move(route);
 }
 
+void ConnectionManagerImpl::ActiveStream::sendLocalReply(
+    ActiveStreamEncoderFilter* filter, bool is_grpc_request, Code code, const std::string& body,
```

Comment:
Oh, I really like where this is going!

My one question for APIs is id we think sendLocalReply should instead take a reference to the request headers so is_grpc_request can be calculated locally in just one place, and in case we eventually end up with other transformations which might be based on request headers.   We can just push this as-is and iterate, but I'd be interested in @mattklein123's take now since I think this is likely to be used in many places here and in downstream filters so it'd be nice to get it right on first pass.

Reply:
@alyssawilk I think this is internal code and not the main API, right? (So I think the code does what you are asking?)

Reply:
Yeah, this is what happens when I do reviews before caffeine :-/

I think we could avoid latching is_grpc_request_ in the two places we do, but that's a smaller request.

Reply:
You mean having `is_grpc_request_` in both ConnManagerImpl and AsyncStreamImpl? Right now I don't have a clear picture of the relation between the two, so I don't know if tracking in one of them can be eliminated or not.